zbus/src/connection/builder.rs

```diff
@@ -525,7 +525,16 @@ impl<'a> Builder<'a> {
             Target::VsockStream(stream) => stream.into(),
             Target::Address(address) => {
                 guid = address.guid().map(|g| g.to_owned().into());
-                match address.connect().await? {
+
+                match address
+                    .clone() //This clone is necessary to capture on error the address because the connect
+                    // function consume the address object.
```

Comment:
We could have the `Address::connect` do the mapping instead, or return `Result<Stream, (io::Error, Self)` (open to other options too) or maybe even have the `Address::connect` just not take ownership of address instance (it's internal after all).

Since this should not require a major effort and there is no urgency here, I think it makes sense to already avoid the cloning here, instead of leaving it for the future, which is likely never going to happen. :)

P.S. do remember the atomic commits rule, so please put the needed refactoring in a separate commit that comes before this one. :pray: